src/assets/Keycloak/oCIS-with-Keycloak.md

```diff
@@ -0,0 +1,184 @@
+> Note: Realm name should name with name provided in oCIS enviroment variable: `OCIS_OIDC_ISSUER: https://host.docker.internal:8443/realms/oCIS`
+
+4. Add a new client:
+![client registration](/src/assets/Keycloak/images/oCIS-Keycloak/client-registration-1.png)
+
+- Client ID: `web`
+![add client](/src/assets/Keycloak/images/oCIS-Keycloak/client-registration-2.png)
+
+- Root URL: `https://host.docker.internal:9200`
+![add root url](/src/assets/Keycloak/images/oCIS-Keycloak/client-registration-3.png)
+
+5. Add realm roles:
+
+- Role name: `ocisAdmin`
+![add realm role](/src/assets/Keycloak/images/oCIS-Keycloak/create-realm-role.png)
+
+> Note: oCIS has defined role name that has been documented under [Automatic Quota Assignments](https://owncloud.dev/services/proxy/#automatic-role-assignments). So, we can't create role with random name.
+
+6. Create a new user:
+
+- Username: `admin` and other info
+![create user](/src/assets/Keycloak/images/oCIS-Keycloak/create-user-1.png)
+![add user information](/src/assets/Keycloak/images/oCIS-Keycloak/create-user-2.png)
+
+- Create password
+![create password](/src/assets/Keycloak/images/oCIS-Keycloak/create-user-3.png)
+![setup password for user](//src/assets/Keycloak/images/oCIS-Keycloak/create-user-4.png)
+
+- Role Mapping: Assign `ocisAdmin` role
+![assign role](/src/assets/Keycloak/images/oCIS-Keycloak/assign-role-to-user-1.png)
+
+![assign role to user](/src/assets/Keycloak/images/oCIS-Keycloak/asign-role-to-user-2.png)
+
+7. Update `roles` Client scope: `Client Scopes` -> `roles`
+- `Mappers` -> `realm roles` -> `Token Claim Name=roles`
+![update-roles](/src/assets/Keycloak/images/oCIS-Keycloak/update-roles.png)
+> Note: oCIS default role_claim is 'roles'
+
+8. Go to url `https://host.docker.internal:9200/` reload if already opened
+![login oCIS with keycloak](//src/assets/Keycloak/images/oCIS-Keycloak/ocis-keycloak-login.png)
+
+### Using JSON files
+Keycloak provides [Importing and Exporting Realms](https://www.keycloak.org/server/importExport) feature. If we have JSON configuration files then we can avoid setting services in keycloak.
+oCIS provides documentation to set up [oCIS with Keycloak](https://owncloud.dev/ocis/deployment/ocis_keycloak/) which uses JSON configuration. oCIS provides source code to deploy
+[oCIS-with-Keycloak](https://github.com/owncloud/ocis/tree/master/deployments/examples/ocis_keycloak).
+
+In this blog post, we've talked about putting Keycloak and oCIS together. We've shown how to set them up using Docker Compose, making sure they work smoothly with Traefik.
+We've also given step-by-step instructions on setting up oCIS with Keycloak manually, like making realms, adding clients and roles, and creating users.
+For those who like using JSON files, Keycloak lets you do that to set things up more easily. By following these tips, you can use Keycloak and oCIS together for better identity and access management in your projects.
```

Comment:
In the conclusion you write that the reader can use the learnings from this blog for his/her own projects, however the examples were very specific to Keycloak and oCIS - it seems like there is a little gap. I assume some experts / experienced users can easily bridge that gap. However, if your target audience are "mid level" developers, then maybe some additional information about the steps above would be helpful; in particular not only explaining what you do but also why this matters. I guess this would help the reader to better understand each step and replicate them in their own projects.